account-service/src/main/java/com/bdonor/accountservice/Models/AccountHelper.java

```diff
@@ -12,44 +13,61 @@
     @Autowired
     private UserRepository UserRepo;
 
-    public User createNew(String firstname){ // This works
-        return UserRepo.save(new User(firstname));
-    }
+    @Autowired
+    private BCryptPasswordEncoder bCryptPasswordEncoder;
 
-    public User create( String bloodGroup, String firstname, String _surname, String _email, String _password, String _addressline, String _postcode ){ // This Works
-        return UserRepo.save( new User(bloodGroup, firstname,  _surname,  _email,  _password,  _addressline,  _postcode) );
+    public User create(String bloodGroup, String firstname, String _surname, String _email, String _password, String _addressline, String _postcode) { // This Works
+        User user = UserRepo.findByEmail(_email);
+        if(user != null) {
+            System.out.println("User Exists");
+            return new User();
+        }
+        return UserRepo.save(new User(bloodGroup, firstname, _surname, _email, bCryptPasswordEncoder.encode(_password), _addressline, _postcode));
     }
 
-    public User getByfirstName(String firstName){
+    public User getByfirstName(String firstName) {
         return UserRepo.findByFirstName(firstName);
     }
 
-    public List<User> getAll(){
+    public List<User> getAll() {
         return UserRepo.findAll();
     }
 
     public User Update(String bloodGroup, String firstname, String _surname, String _email, String _password, String _addressline, String _postcode) {
```

Comment:
parameters inconsistent, let's use "_" in class variables only (it's only naming conventions but goes well with Java)